Assignment2-yixiaoli.Rmd

```diff
@@ -0,0 +1,147 @@
+names(V.STD) <- c("id", "gender", "topic", "group")
+
+#Bind the two data frames together (you will get a warning because the teacher data frame has 5 types of id (A,B,C,D,E) and the student has 25 (1-30), this isn't a problem)
+VERTEX <- bind_rows(V.TCH, V.STD)
+
+```
+
+Now we have both a Vertex and Edge list it is time to plot our graph!
+
+```{r}
+#Load the igraph package
+
+library(igraph)
+
+#First we will make an object that contains the graph information using our two dataframes EDGE and VERTEX. Notice that we have made "directed = TRUE" - our graph is directed since discipline is being given from a teacher to a student.
+
+g <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX)
+
+#Now we can plot our graph using the force directed graphing technique - our old friend Fruchertman-Reingold!
+
+plot(g,layout=layout.fruchterman.reingold)
+
+#There are many ways to change the attributes of the graph to represent different characteristics of the newtork. For example, we can color the nodes according to gender.
+
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender)
+
+#We can change the thickness of the edge according to the number of times a particular teacher has sent a particular student to the principal.
+
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count)
+
+````
+
+## Part II
+
+In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
+```{r}
+library(igraph)
+D1 <- read.csv("discipline-data.csv")
+library(dplyr)
+D2 <- select(D1, tid, stid)
+
+# count the number of actions per student
+STUDENTCOUNT <- count(D1, stid)
+names(STUDENTCOUNT) <- c("id", "count")
+
+# count the number of actions per teacher
+TEACHERCOUNT <- count(D1, tid)
+names(TEACHERCOUNT) <- c("id", "count")
+
+# combine the counts together
+actions <- rbind(STUDENTCOUNT,TEACHERCOUNT)
+
+# modify the vertex size with the counts
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count, vertex.size=actions$count)
+```
+
+## Part III
+
+Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
+
+Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweenness centrality and degree. **Who is the most central person in the network?**
+```{r}
+library(tidyr)
+
+D3 <- read.csv('hudk4050-classes-new.csv', na.strings=c("","NA"))
+D4 <- D3 %>% gather(Class, ClassID, Class.1:Class.6, na.rm = TRUE)
+
```

Comment:
You are on the right track but you are missing a step, which is cleaning the dataset. Some people have put a space between the department and the course number and some haven't; thus, these values will be treated differently.